### DISCUSSION TITLE
cursor on function's beginning when `Goto next function`?

### DISCUSSION THREAD

**Discussion by Luv-Ray** [helix-editor/helix#9997] [Ideas]
Sometimes there is a long function, and when I run `Goto next function`, the cursor will be at the end of the funciton and I can't see the fuction's head. So I have to run `Goto matching bracket` to see which function it is. Is it possible to have a configuration option that allows `Goto next function` jump to the beginning of a function?
Also `Goto next type definition` please.

**Comment by kirawi** [comment]
AFAIK this is controlled by the LSP not Helix.

**Comment by Luv-Ray** [comment]
@kirawi Hello, I have implemented it by modifying Helix, this will affect `Goto next function`, `Goto next type definition`, `Goto next parameter` and `Goto next comment`. All of them will keep the cursor at the beginning.
I want to know if I should create a PR now or discuss it here first. Thanks!

**Reply by the-mikedavis** [reply]
I would be interested in a PR that changes the directions when using `]f`/`]c`/etc following https://github.com/helix-editor/helix/pull/8875#discussion_r1428152259. When in select mode, extending so that the cursor is at the end is important so that you can grow the selection. But when using those commands in normal mode we could use `Direction::Backwards` for each range so that the cursor is at the beginning of the selection.
